contracts/FeralfileEnglishAuction.sol

```diff
@@ -0,0 +1,378 @@
+    }
+
+    function auctionOngoing(bytes32 aucId_) public view returns (bool) {
+        // check the auction status and return
+        require(
+            auctions[aucId_].id != 0,
+            "FeralfileEnglishAuction: auction not found"
+        );
+
+        return
+            auctions[aucId_].startAt <= block.timestamp &&
+            auctions[aucId_].endAt >= block.timestamp;
+    }
+
+    function setEnglishAuctions(
+        EnglishAuction[] calldata auctions_
+    ) external onlyOwner {
+        // for-loop to set the auction
+        // able to reset an auction if there's no winner and highest bid = 0
+        for (uint256 i = 0; i < auctions_.length; i++) {
+            EnglishAuction memory auc_ = auctions[auctions_[i].id];
+
+            require(
+                auc_.id == 0,
+                "FeralfileEnglishAuction: auction already exist"
+            );
+
+            require(
+                auctions_[i].startAt >= block.timestamp,
+                "FeralfileEnglishAuction: auction start time should be in the future"
+            );
+
+            require(
+                auctions_[i].endAt > auctions_[i].startAt,
+                "FeralfileEnglishAuction: auction end time should be after start time"
+            );
+
+            require(
+                auctions_[i].extendDuration >= 0,
+                "FeralfileEnglishAuction: auction extend duration should be greater than or equal 0"
+            );
+
+            require(
+                auctions_[i].extendThreshold >= 0,
+                "FeralfileEnglishAuction: auction extend threshold should be greater than or equal 0"
+            );
+
+            require(
+                auctions_[i].minIncreaseFactor > 0,
+                "FeralfileEnglishAuction: auction min increase factor should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minIncreaseAmount > 0,
+                "FeralfileEnglishAuction: auction min increase amount should be greater than 0"
+            );
+
+            require(
+                auctions_[i].minPrice > 0,
+                "FeralfileEnglishAuction: auction min price should be greater than 0"
+            );
+
+            auctions[auctions_[i].id] = auctions_[i];
+        }
+    }
+
+    function listAuctionLatestStatus(
+        bytes32[] memory aucIds_
+    ) external view returns (LatestAuctionStatus[] memory) {
+        LatestAuctionStatus[] memory results = new LatestAuctionStatus[](
+            aucIds_.length
+        );
+        for (uint i = 0; i < aucIds_.length; i++) {
+            // get the auction object
+            EnglishAuction memory auction_ = auctions[aucIds_[i]];
+            // get the last auction bid object
+            results[i].latestBid = auctionLatestBid[aucIds_[i]];
+            results[i].endAt = auction_.endAt;
+            results[i].isSettled = auction_.isSettled;
+        }
+        return results;
+    }
+
+    function isValidBidRequest(
```

Comment:
I'd suggest changing function name to `isValidNewBid`

Reply:
@tienngovan Any concern here? We. could figure a better name, i dont think `isValidBidRequest` is proper.